src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -3058,10 +3059,25 @@ class AccessStatementSegment(BaseSegment):
                 Sequence("ALL", "SCHEMAS", "IN", "DATABASE"),
                 Sequence("FUTURE", "SCHEMAS", "IN", "DATABASE"),
                 _schema_object_types,
-                Sequence("ALL", _schema_object_types_plural, "IN", "SCHEMA"),
+                Sequence(
```

Comment:
@tunetheweb: Any thoughts on these changes? I had a couple of potential concerns:
* `_schema_object_types_plural` is already a `OneOf()`. Below, we're embedding it in another `OneOf()`. Seems strange to nest `OneOf()` like this.
* I wonder if this is adding too much Snowflake-specific stuff to ANSI. (It's a judgement call, I know.)

Reply:
I don't see an issue with the `OneOf` thing. It's not a `OneOf` here, just the first part is (whether to use `TEMP` or `TEMPORARY` before the mandatory `TABLE`.

```py
Sequence(OneOf("TEMP", "TEMPORARY"), "TABLE"),
```

The second question is a good one. Particularly adding new keywords to `ANSI` can mean bad error messages for other dialects when they get unparsable code, as it tries to go into these segments and then complains it doesn't know what `PIPE` keyword is. Let me dig into it a little more...

Reply:
Re: `OneOf()`, I was referring to lines 3064-3069 and lines 3075-3080, which look like they are nesting `_schema_object_types_plural` in a `OneOf()`.

And `_schema_object_types_plural` is defined as:
```
    _schema_object_types_plural = OneOf(
        *[f"{object_name}S" for object_name in _schema_object_names]
    )
```

Reply:
actually, there was the similar code of OneOf(..., OneOf(),....) in the code before my changes (lines 3005-3009). For example, 
```
OneOf(
                    "SCHEMA",
                    Sequence("MASKING", "POLICY"),
                    "PIPE",
                    _schema_object_types,
                ),
...
```
where:
```
_schema_object_types = OneOf(
        *_schema_object_names,
        Sequence("MATERIALIZED", "VIEW"),
        Sequence("EXTERNAL", "TABLE"),
        Sequence("FILE", "FORMAT"),
    )
```
I can change my code, of course. But I saw the similar code was using previously, so I used it too. And checked that it's working properly

Reply:
@barrywhart Could you, please, tell me what the problem with checks could be? All other checks are passed, but  the last one is "codecov/project Expected — Waiting for status to be reported" for a long period of time. Tried to google the issue, made an empty commit but it didn't help

Reply:
I think the nested OneOf is still fine btw.

Reply:
@tunetheweb I moved extended grant syntax to the snowflake dialect and added some more options to it

Reply:
Oh excellent!